build/accelerate_launch.py

```diff
@@ -20,23 +20,79 @@
 # Standard
 import os
 import logging
+import subprocess
+import sys
+import traceback
 
 # Third Party
 from accelerate.commands.launch import launch_command
 
 # Local
-from build.utils import process_accelerate_launch_args, get_job_config
+from build.utils import (
+    process_accelerate_launch_args,
+    get_job_config,
+    write_termination_log,
+    USER_ERROR_EXIT_CODE,
+    INTERNAL_ERROR_EXIT_CODE,
+)
 
 
 def main():
     LOGLEVEL = os.environ.get("LOG_LEVEL", "WARNING").upper()
     logging.basicConfig(level=LOGLEVEL)
 
-    job_config = get_job_config()
+    ##########
+    #
+    # Parse arguments
+    #
+    ##########
+    try:
+        job_config = get_job_config()
 
-    args = process_accelerate_launch_args(job_config)
-    logging.debug("accelerate launch parsed args: %s", args)
-    launch_command(args)
+        args = process_accelerate_launch_args(job_config)
+        logging.debug("accelerate launch parsed args: %s", args)
+    except FileNotFoundError as e:
+        logging.error(traceback.format_exc())
+        write_termination_log("Unable to load file: {}".format(e))
+        sys.exit(USER_ERROR_EXIT_CODE)
+    except (TypeError, ValueError, EnvironmentError) as e:
+        logging.error(traceback.format_exc())
+        write_termination_log(
+            "Exception raised during training. This may be a problem with your input: {}".format(
+                e
+            )
+        )
+        sys.exit(USER_ERROR_EXIT_CODE)
+    except Exception as e:  # pylint: disable=broad-except
+        logging.error(traceback.format_exc())
+        write_termination_log("Unhandled exception during training")
+        sys.exit(INTERNAL_ERROR_EXIT_CODE)
+
+    ##########
+    #
+    # Launch training
+    #
+    ##########
+    try:
+        launch_command(args)
+    except subprocess.CalledProcessError as e:
+        # If the subprocess throws an exception, the base exception is hidden in the subprocess call
+        # and is difficult to access at this level. However, that is not an issue because
+        # launch_training.py would have already written the exception message to termination log.
+        logging.error(traceback.format_exc())
+        # The exit code that launch_training.py threw is captured in e.returncode
+
+        return_code = e.returncode
+        if return_code not in [INTERNAL_ERROR_EXIT_CODE, USER_ERROR_EXIT_CODE]:
+            return_code = INTERNAL_ERROR_EXIT_CODE
+            write_termination_log("Unhandled exception during training")
+        sys.exit(return_code)
+    except Exception as e:  # pylint: disable=broad-except
```

Comment:
Just as a brain dump of errors I've hit into....an error I could see happening here is if the user passes in a model at a wrong path, the code will try to load that model from huggingface which leads to error:
```
HFValidationError(
huggingface_hub.utils._validators.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': '/dccstor/anhtest/model_output/tuning/llama2-7b-twitter-lora-default-flash-checkpoint-15-merged-adapter/'. Use `repo_type` argument if needed.
```

Reply:
Perfect, catching that one now too.